src/super_gradients/common/sg_loggers/base_sg_logger.py

```diff
@@ -348,3 +352,27 @@ def _save_code(self):
             self.add_file(name)
             code = "\t" + code
             self.add_text(name, code.replace("\n", "  \n  \t"))  # this replacement makes tb format the code as code
+
+    def _validate_checkpoint(self, state_dict: Mapping, path=None) -> None:
+        """
+        Validate the checkpoint state_dict to make sure it contains only primitive types and tensors.
+        This function raises ValueError if the checkpoint contains ListConfig or DictConfig.
+
+        :param state_dict:  Checkpoint state_dict.
+        :param path:        Indicates the current path of the checkpoint
+                            (Used to print meaningful path if problematic key is detected)
+        """
+        if isinstance(state_dict, (ListConfig, DictConfig)):
+            raise ValueError(
+                f"Checkpoint state_dict element {path} contain ListCongfig and DictConfig."
+                f"Only types and Tensors are supported."
+                f"Most likely, you forgot to convert those to container using OmegaConf.to_container()."
+            )
```

Comment:
Do we really want to raise an error ?
It would suck if you train a model just to get an error when saving your checkpoint.
This also means that if we somehow introduce a bug in SG (or in a client delivery) where we save Listconfig, the model wouldn't be able to train at all without fixing.
Maybe we want this behavior because people often don't read warnings, but I think it's worth thinking twice about it, I dont know if you thought about that

Reply:
Fixed in 4fa008e6717e1d0863d481372ef7ab418dd9c2a3